### ISSUE TITLE
Calculate best mask dtype from max label-value

### ISSUE THREAD

**Issue by will-moore** [ome/omero-cli-zarr#54]
Currently, the default `label_bits` is set to 64, which is much higher than is needed in many cases.
We could calculate a more suitable value from the highest label-value we need to store.

```
 masks.add_argument(
            "--label-bits",
            default=str(max(MASK_DTYPE_SIZE.keys())),
            choices=[str(s) for s in sorted(MASK_DTYPE_SIZE.keys())],
            help=(
                "Integer bit size for each label pixel, use 1 for a binary "
                "label, default %(default)s"
            ),
        )
```

```
MASK_DTYPE_SIZE: Dict[int, np.dtype] = {
    1: np.bool,
    8: np.int8,
    16: np.int16,
    32: np.int32,
    64: np.int64,
}
```
cc @joshmoore

**Comment by joshmoore** [comment]
:+1: This may also trigger one or more additions to the spec:
 - "one SHOULD use the smallest value that's reasonable (incl. unsigned)"
 - "one MUST make sure that all dtypes are the same across the multiscales"

**Comment by sbesson** [comment]
Is that also fixed by #116 @will-moore? What are the implications for the `--label-bit` option? Should the description be minimally updated to say the default will auto-compute or should the flag be dropped completely?

**Comment by will-moore** [comment]
yes, this is closed by #116

As far as I can see from the code, `args.label_bits` isn't actually used for the output labels dtype (at least for masks - might be used for Polygons -> labels).
Haven't tested yet.
I didn't add that option, so I'm not sure of the use-case, or if it's still needed.

**Comment by sbesson** [comment]
I gave it a try both with `omero zarr masks --label-bits` and `omero polygon masks --label-bits` but it looks like the argument was not recognized in `omero-cli-zarr 0.3.0` (where the labels were unconditionally stored as `int64`) and with the changes in #116 the mask `dtype` is automatically computed based on the label dynamic range.

Internally, it looks like the value is passed to `MaskSaver` and saved as an attribute and then used in the `_mask_to_binim_yx` and `_polygon_to_binim_yx` to determine the type of `np.array`.

Unless @joshmoore or someone else remembers the use case, I would propose to remove the no-op argument. There is potentially a follow-up action to clean-up the internal use of `dtype`.
